### PR TITLE
Add option to disable sessionState in Accordion

### PR DESCRIPTION
**Pull request by querkmachine**
Adds a new configuration option to the Accordion component, allowing users to explicitly disable (or enable) the session storage and restoration functionality.

Some developers have expressed a desire to disable this function, which currently can only be done by separately clearing `sessionStorage` or randomising the `id` of accordions, as described in #3182.

## Option naming

My understanding of good naming is that boolean options, like checkboxes, should always be labelled using affirmative, opt-in kind of language: yes, remember my password; yes, allow analytics cookies; yes, I want disposable cutlery; and so forth.

As such, the parameter is named affirmatively—`rememberExpanded` in JS and Nunjucks APIs and `data-remember-expanded` in the HTML API.

I opted for this name as it's shorter and more self-explanatory (or maybe, less technical-sounding) than others I thought of (e.g. `rememberExpandedSections`, `rememberSectionState`, `enableStateRemembering`), but I'm not tied to it and am open to suggestions. It also feels more congruent with the existing `expanded` option available to each section.

However, this means that the default value for this setting is `true` (enabled) with the typical use case being developers setting it to `false` (disabled), which might be a bit unintuitive. 

## Usage

Like internationalisation options, the configuration can be provided via JavaScript initialisation, in HTML using a `data-*` attribute, or via a parameter on the `govukAccordion` Nunjucks macro.

JavaScript example:

```js
new window.GOVUKFrontend.Accordion(document.getElementById('accordion'), {
  rememberExpanded: false
}).init()
```

Nunjucks example:
```nunjucks
{{ govukAccordion({
  id: "accordion",
  rememberExpanded: false,
  items: […]
)} }}
```

HTML example:
```html
<div class="govuk-accordion" data-module="govuk-accordion" id="accordion" data-remember-expanded="false">
 […]
</div>
```

## Changes

- The Accordion's default configuration has a new default: `rememberExpanded: true`.
- The `setInitialState` and `storeState` methods of the Accordion's JS now check that `rememberExpanded` is `true` in addition to ensuring that `sessionState` is available.
- The Accordion template now looks for and uses `rememberExpanded` if it's provided. 
- Added documentation for the `rememberExpanded` Nunjucks parameter. Also tweaked the `id` parameter documentation.
- Added example and tests for the existence of the `data-*` attribute in the template and for the functionality itself.

### CODE REVIEW

**Review by colinrotherham — APPROVED** [review]
This looks ace 🙌 

No code comments to add

I'm perhaps torn on whether `.restoreExpanded()`  would be better?

But I've had had `jest.restoreAllMocks()` on my mind

See what @claireashworth thinks

File: CHANGELOG.md [hunk 1]
```diff
@@ -31,6 +31,16 @@ Only use disabled form controls if research shows it makes the user interface ea
 
 This was added in [pull request #3187: Add disabled styles for form controls](https://github.com/alphagov/govuk-frontend/pull/3187).
 
+#### Stop the Accordion from remembering and restoring sessions
+
+By default, the [Accordion](https://design-system.service.gov.uk/components/accordion/) will remember which sections a user has expanded and collapsed upon leaving a page. If the user later returns to the page, this layout will be restored, overriding any sections that have been manually set as `expanded` in code.
```
Comment (claireashworth) [inline CHANGELOG.md:36]:
```suggestion
By default, when a user leaves a page, the [Accordion](https://design-system.service.gov.uk/components/accordion/) will remember the layout of expanded and collapsed sections selected by the user. If the user returns to the page, this layout will be restored and override any sections manually set as `expanded` in code.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -31,6 +31,16 @@ Only use disabled form controls if research shows it makes the user interface ea
...
+
+You can now disable this functionality by using the `rememberExpanded` option in the `govukAccordion` Nunjucks macro.
```
Comment (claireashworth) [inline CHANGELOG.md:38]: Do we need to add a link to the relevant nunjucks macro?

Comment (querkmachine) [reply CHANGELOG.md:38]: I don't think so. The user-facing version of the Nunjucks macro code is already part of the design system page linked in the previous paragraph.

Comment (claireashworth) [reply CHANGELOG.md:38]: ok great

File: CHANGELOG.md [hunk 3]
```diff
@@ -31,6 +31,16 @@ Only use disabled form controls if research shows it makes the user interface ea
 
 This was added in [pull request #3187: Add disabled styles for form controls](https://github.com/alphagov/govuk-frontend/pull/3187).
 
+#### Stop the Accordion from remembering and restoring sessions
```
Comment (36degrees) [inline CHANGELOG.md:34]: A thought – we might want to reword this as we already have a few other entries in the changelog that start with stop where we _want_ people to stop doing something, but in this instance we're just _allowing_ people to disable it where it's not desired behaviour:

- [We want you to] stop using the deprecated IE8 mixins and settings
- [We want you to] stop using the `govuk-button--disabled` class on buttons
- [You now have the option to] stop the Accordion from remembering and restoring sessions

Maybe something like 'Configure whether the Accordion remembers and restores sessions'?

**Review by claireashworth — APPROVED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (querkmachine) [reply CHANGELOG.md]: Makes sense to me! 

I'm working under the assumption that the changelog is going to get tidied up a bit pre-release anyway, but no harm in making things clearer now.

**Comment by querkmachine** [comment]
There doesn't seem to be any strong opinions regarding the parameter name, so I'm going to keep it as-is for now. We still have a little bit of leeway before release to change it if we want to.

**Comment by querkmachine** [comment]
I'm now having second thoughts on the parameter naming. Having noticed that many other options in Frontend don't follow the 'affirmative language' guideline—like the Button's`preventDoubleClick` and Error Summary's `disableAutoFocus`—it may be more consistent to reverse the logic of this one too.
